tools/clang/test/CodeGenHLSL/declarations/functions/overloading/namespace_qualified_no_intrinsic_candidate.hlsl

```diff
@@ -0,0 +1,19 @@
+// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s
+
+// Regression test for GitHub #1884, where intrinsics were considered
+// valid overload candidates for overloaded functions of the same name in a namespace
+
+// CHECK: call void @dx.op.storeOutput.i32(i32 5, i32 0, i32 0, i8 0, i32 42)
+
+namespace foo
+{
+    int abs(int2 x) { return 42; }
+    int abs() { return 0; }
```

Comment:
if you defined this function, then better to call this one as well in main and add a CHECK for it.

Reply:
Hmm, what more would that be testing?

Reply:
I'm confused, main is what emits storeOutput, not the overloads...? There is certainly a need for overload tests, but that's a task of its own.

Reply:
Ok. Btw, this is what I meant:

```
// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s

// Regression test for GitHub #1884, where intrinsics were considered
// valid overload candidates for overloaded functions of the same name in a namespace

// CHECK: call void @dx.op.storeOutput.i32(i32 5, i32 0, i32 0, i8 0, i32 42)
// CHECK: call void @dx.op.storeOutput.i32(i32 5, i32 0, i32 0, i8 0, i32 0)

namespace foo
{
    int abs(int2 x) { return 42; }
    int abs() { return 0; }
}

int2 main() : OUT
{
    // This should not consider the abs(int) intrinsic from the global namespace,
    // regardless of the fact that it is a better match for the arguments.
    return int2(foo::abs(0), foo::abs());
}
```

Reply:
My last commit includes this.